crates/notion-core/src/tool/mod.rs

```diff
@@ -60,15 +81,49 @@ impl ToolSpec {
     pub fn uninstall(&self, session: &mut Session) -> Fallible<()> {
         match self {
             ToolSpec::Node(_version) => unimplemented!("Uninstalling Node not supported yet"),
-            ToolSpec::Yarn(_version) => unimplemented!("Uninstalling Yarn not supported yet"),
             // ISSUE(#292): Implement install for npm
             ToolSpec::Npm(_version) => unimplemented!("Uninstalling Npm not supported yet"),
+            ToolSpec::Yarn(_version) => unimplemented!("Uninstalling Yarn not supported yet"),
             ToolSpec::Package(name, _version) => {
                 session.uninstall_package(name.to_string())?;
             }
         }
         Ok(())
     }
+
+    /// Try to parse a tool and version from a string like `<tool>[@<version>].
+    pub fn try_from_str(tool_spec: &str) -> Fallible<Self> {
+        let captures =
+            TOOL_SPEC_PATTERN
+                .captures(tool_spec)
+                .ok_or(ErrorDetails::ParseToolSpecError {
+                    tool_spec: tool_spec.into(),
+                })?;
+
+        // Validate that the captured name is a valid NPM package name.
+        let name = &captures["name"];
+        if let Validity::Invalid { errors, warnings } = validate(name) {
+            dbg!(errors);
+            dbg!(warnings);
```

Comment:
Should these debugging prints be removed? It looks from [the docs](https://doc.rust-lang.org/std/macro.dbg.html) that it isn't removed on production builds, it's primarily a shortcut for `eprintln!("{:?}", expr)`

Reply:
Oh, yes! Thank you, and good catch!